Replace backup link with updated url, Fix backup link logic

adamw3455 opened the pull request:
Closes #23 

Deployed at https://courses-dev-adamw3455-footer-link.dokku-03.cs.ucsb.edu/

The previous footer backup source code link was https://github.com/ucsb-cs156-f22/f22-5pm-courses, which is an outdated, quarter specific link. I corrected it to the main courses repo https://github.com/ucsb-cs156/proj-courses.

The footer also did not correctly render the backup source code link, which I fixed by improving the undefined/missing sourceRepo detection logic.

Screenshot of correct HTML component/href attribute with blocked systemInfo API call:
<img width="919" height="519" alt="image" src="https://github.com/user-attachments/assets/51223754-a19c-4538-8c99-69d75a8177e4" />

divyanipunj commented:
I think that this addresses the feature described. However, when I tested it out by blocking systemInfo, it doesn't render any link for the GitHub. While this bug is not a product of your code, we can't merge in broken code. Please make sure that a link actually renders when you block systemInfo. 

You can test it out by blocking request URL and checking if the link renders.
<img width="1512" height="804" alt="Screenshot 2025-11-20 at 6 08 49 PM" src="https://github.com/user-attachments/assets/d311eb9a-72cc-4d40-841e-6f2cebfec4d9" />

adamw3455 commented:
> I think that this addresses the feature described. However, when I tested it out by blocking systemInfo, it doesn't render any link for the GitHub. While this bug is not a product of your code, we can't merge in broken code. Please make sure that a link actually renders when you block systemInfo.
> 
> You can test it out by blocking request URL and checking if the link renders. <img alt="Screenshot 2025-11-20 at 6 08 49 PM" width="1512" height="804" src="https://private-user-images.githubusercontent.com/183317941/517172243-d311eb9a-72cc-4d40-841e-6f2cebfec4d9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM3NzIxMTQsIm5iZiI6MTc2Mzc3MTgxNCwicGF0aCI6Ii8xODMzMTc5NDEvNTE3MTcyMjQzLWQzMTFlYjlhLTcyY2MtNGQ0MC04NDFlLTZmMmNlYmZlYzRkOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyMlQwMDM2NTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMDU4MDhhNzBkMDIzOTgwZjEyZDlmZjg0ZTcwZDQ4ZjBlZTQzODc1NTM2MDI2MzczNTlkMGUxOWJlMzhiZGM5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.dpLTeh5biMIWl68UdrzIcEcJgyTEa7BmHzDb0loIdFg">

Got it. I updated the logic for displaying the backup link, and added some tests to verify that it adds the correct href when systemInfo is blocked/undefined/etc. You can test on the dev deployment, or just see the screenshot in PR description.
